core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -565,13 +568,24 @@ public synchronized void checkpoint() throws IOException {
    * The caller is responsible for detecting and responding to leadership changes.
    */
   private void catchUp(JournalStateMachine stateMachine, LocalFirstRaftClient client)
-      throws TimeoutException, InterruptedException {
+          throws TimeoutException, InterruptedException {
     long startTime = System.currentTimeMillis();
     long waitBeforeRetry = ServerConfiguration.global()
         .getMs(PropertyKey.MASTER_EMBEDDED_JOURNAL_CATCHUP_RETRY_WAIT);
     // Wait for any outstanding snapshot to complete.
     CommonUtils.waitFor("snapshotting to finish", () -> !stateMachine.isSnapshotting(),
         WaitForOptions.defaults().setTimeoutMs(10 * Constants.MINUTE_MS));
+    OptionalLong endCommitIndex = OptionalLong.empty();
+    try {
+      endCommitIndex = getGroupInfo().getCommitInfos().stream()
+          .mapToLong(RaftProtos.CommitInfoProto::getCommitIndex).min();
```

Comment:
Do we choose minimum commit index as an approximation, the set of quorum acknowledged messages can be further than the min commit index?

Reply:
This is the minimum commit index of all the maximum commits indexes which exist between each journal in the quorum. I opted for the lower bound. It's possible that some nodes in the quorum may have a higher commit index. I think min() should still give a good approximation because we don't expect nodes in the quorum to be many commits behind? I actually don't know the behavior of the system if that occurs.

We can change this to `max()` if you feel using the pure maximum bound of all journals is more appropriate. I don't have an opinion either way

Reply:
One case might affect the estimation is when there is a new master added to the quorum. The new node will probably get zero commit index initially and catch up slowly to the leader.

Would it be possible to use the current leader commit index? An elected leader is guaranteed to have the latest commit of the quorum.

Reply:
That's a good point. I will look into whether or not I can get the commit index for the current leader. I tried finding this but if I recall the API surface is quite limited getting information out of ratis which is why I've implemented this way.

If I can't find a way to get the current leader index I'll change this to max() to use the upper-bound for time estimation

Reply:
I addressed this issue in the latest commit. You can see the new logic here:
https://github.com/Alluxio/alluxio/pull/13107/commits/99fe56c3510e89a8085f7270acaa5c46d1ae9f02#diff-37edc9788907ad25cdb171ff28732c44ba2bc66325be1a97f573b71beadfd092R585